magellan-library/src/test/java/com/wealthfront/magellan/navigation/NavigatorTest.kt

```diff
@@ -0,0 +1,150 @@
+    assertThat(navigator.backStack.first.direction).isEqualTo(FORWARD)
+    assertThat(navigator.backStack.first.navigationType).isEqualTo(SHOW)
+  }
+
+  @Test
+  fun replaceGo() {
+    navigator.show(navigable1)
+    navigator.replaceAndGo(navigable2)
+
+    assertThat(navigator.backStack.size).isEqualTo(1)
+    assertThat(navigator.backStack.first.navigable).isEqualTo(navigable2)
+    assertThat(navigator.backStack.first.direction).isEqualTo(FORWARD)
+    assertThat(navigator.backStack.first.navigationType).isEqualTo(GO)
+  }
+
+  @Test
+  fun replaceShow() {
+    navigator.show(navigable1)
+    navigator.replaceAndShow(navigable2)
+
+    assertThat(navigator.backStack.size).isEqualTo(1)
+    assertThat(navigator.backStack.first.navigable).isEqualTo(navigable2)
+    assertThat(navigator.backStack.first.direction).isEqualTo(FORWARD)
+    assertThat(navigator.backStack.first.navigationType).isEqualTo(SHOW)
+  }
+
+  @Test
+  fun goBack_multipleScreen_removeScreenFromBackstack() {
+    navigator.goTo(navigable1)
+    navigator.goTo(navigable2)
+    val didNavigate = navigator.goBack()
+
+    assertThat(didNavigate).isTrue()
+    assertThat(navigator.backStack.size).isEqualTo(1)
+    assertThat(navigator.backStack.first.navigable).isEqualTo(navigable1)
+    assertThat(navigator.backStack.first.direction).isEqualTo(FORWARD)
+    assertThat(navigator.backStack.first.navigationType).isEqualTo(GO)
+  }
+
+  @Test
+  fun goBack_oneScreen_backOutOfApp() {
+    navigator.goTo(navigable1)
+    val didNavigate = navigator.goBack()
+
+    assertThat(didNavigate).isFalse()
+    assertThat(navigator.backStack.size).isEqualTo(1)
+    assertThat(navigator.backStack.first.navigable).isEqualTo(navigable1)
+    assertThat(navigator.backStack.first.direction).isEqualTo(FORWARD)
+    assertThat(navigator.backStack.first.navigationType).isEqualTo(GO)
+  }
+
+  @Test
+  fun historyRewriter() {
+    navigator.navigate(FORWARD) {
+      it.push(NavigationEvent(navigable1, FORWARD, GO))
+      it.push(NavigationEvent(navigable2, FORWARD, GO))
+      it.push(NavigationEvent(navigable3, FORWARD, SHOW))
+    }
+
+    assertThat(navigator.backStack.size).isEqualTo(3)
+    assertThat(navigator.backStack.first.navigable).isEqualTo(navigable3)
+    assertThat(navigator.backStack.first.direction).isEqualTo(FORWARD)
+    assertThat(navigator.backStack.first.navigationType).isEqualTo(SHOW)
+  }
+
+  @Test
+  fun goBack_withoutScreen_backOutOfApp() {
+    val didNavigate = navigator.goBack()
+
+    assertThat(didNavigate).isFalse()
+  }
+
+  @Test
+  fun getScreenTraversal() {
+    navigator.goTo(navigable1)
+    navigator.goTo(navigable2)
+    navigator.goTo(navigable3)
+
+    assertThat(navigable1.getScreenTraversal()).isEqualTo(listOf(navigable1, navigable2, navigable3))
+  }
+
+  @Test
+  fun getTraversalDescription() {
+    navigator.goTo(navigable1)
+    navigator.goTo(navigable2)
+    navigator.goTo(navigable3)
+
+    assertThat(navigable1.getTraversalDescription())
```

Comment:
Will add a androidTest when i introduce a sample to test the traversal of a multi-navigator description.